### ISSUE TITLE
Generalize `linalg.cholesky` beyond 2D arrays

### ISSUE THREAD

**Issue by ricardoV94** [aesara-devs/aesara#791]
```python
import numpy as np
import aesara.tensor as at

x = np.full((2, 3, 3), np.eye(3))
np.linalg.cholesky(x)
at.linalg.cholesky(x) # Assertion error x.ndim == 2
```

**Comment by Sayam753** [comment]
Hi
I have started to look into this issue to make progress on https://github.com/pymc-devs/pymc/pull/5424. The source code of Cholesky Op defines a L_Op method:

https://github.com/aesara-devs/aesara/blob/da86d351979c0a29fc80da16db965551e9e8c14f/aesara/tensor/slinalg.py#L70

Do we need to update this method as well alongside changing `perform` method?

**Comment by brandonwillard** [comment]
> To generalize `linalg.cholesky` beyond 2d arrays, do we need to update this method as well alongside changing `perform` method?

Yes, as well as `Op.infer_shape` and everything else.  It's always possible to raise exceptions in those cases, but that would make the `Op`/new capability much less useful.

**Comment by Sayam753** [comment]
@lucianopaz no, feel free to work on the issue. I am still wrapping my head around how the grad operations in Op work.

**Comment by ricardoV94** [comment]
@lucianopaz The perform method of the Op is free to do whatever it wants, including Python loops across batches, or just call a `np.vectorized` version of the scipy `solve` if we have reason to believe that one is better anyway.
